app/models/billing_expense.rb

```diff
@@ -11,12 +13,16 @@ class BillingExpense < ApplicationRecord
 
   validates :amount, inclusion: { in: AMOUNT }
 
-  def billing_hours
-    relevant_hours = Hour.where(billing_expense: id)
-    relevant_hours.sum(&:hours) + relevant_hours.sum(&:minutes) / 60
-  end
-
-  def billed_hours
-    Hour.where(billing_expense: id)
+  def compute_amount
+    hour_count = id ? hours.total_hours : volunteer.hours.billable.total_hours
```

Comment:
Regarding the `id ?`, if you want to check if a record was saved there's also `persisted?` (no need to change it here, just FYI)